sdk/python/runtime/main.go

```diff
@@ -24,44 +35,57 @@ const (
 //go:embed scripts/runtime.py
 var runtimeTmpl string
 
-func (m *PythonSdk) ModuleRuntime(modSource *Directory, subPath string, introspectionJson string) *Container {
-	return m.CodegenBase(modSource, subPath, introspectionJson).
+func (m *PythonSdk) ModuleRuntime(
+	ctx context.Context,
+	modSource *ModuleSource,
+	introspectionJson string,
+) (*Container, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, err
+	}
+
+	return ctr.
 		WithExec([]string{"python", "-m", "pip", "install", "."}).
-		WithWorkdir(ModSourceDirPath).
 		WithNewFile(RuntimeExecutablePath, ContainerWithNewFileOpts{
 			Contents:    runtimeTmpl,
 			Permissions: 0755,
 		}).
-		WithEntrypoint([]string{RuntimeExecutablePath})
+		WithEntrypoint([]string{RuntimeExecutablePath}), nil
 }
 
-func (m *PythonSdk) Codegen(modSource *Directory, subPath string, introspectionJson string) *GeneratedCode {
-	ctr := m.CodegenBase(modSource, subPath, introspectionJson)
+func (m *PythonSdk) Codegen(ctx context.Context, modSource *ModuleSource, introspectionJson string) (*GeneratedCode, error) {
+	ctr, err := m.CodegenBase(ctx, modSource, introspectionJson)
+	if err != nil {
+		return nil, err
+	}
+
 	ctr = ctr.WithDirectory(genDir, ctr.Directory(sdkSrc), ContainerWithDirectoryOpts{
 		Exclude: []string{
 			"**/__pycache__",
 		},
 	})
 
 	modified := ctr.Directory(ModSourceDirPath)
-	diff := modSource.Diff(modified).Directory(subPath)
+	diff := modSource.RootDirectory().Diff(modified)
```

Comment:
```suggestion
	diff := modSource.Directory(".").Diff(ctr.Directory("."))
```

Reply:
The rule for SDKs is that they need to return the diff of the module source's root directory. This is needed for cases where we may end up needing to modify files outside the root (i.e. `go.mod` maybe, for the Go SDK). But then because some SDKs need to, we want all SDKs to follow this rule for consistency.

Reply:
This was due to my "meta" confusion. 😅

To enforce the consistency rule, could we move the diff to the engine?

Reply:
Yeah good call, did this and also realized along the way that the diff wasn't even serving it's intended purpose previously (SDKs we're essentially doing `B - A` but then the engine was merging the diff back onto the original directory, so `A + (B - A) == B`). So fixed that now too, only export the minimized diffs back to clients.